From d5d225d1567cc7bb2e0363f077e7f2a1abc78afb Mon Sep 17 00:00:00 2001
From: Saurabh Sengar <saurabh.singh@xilinx.com>
Date: Fri, 23 Feb 2018 23:51:30 +0530
Subject: [PATCH 0289/1852] drm: xlnx: mixer: Correcting the cursor width and
 height

commit 9539c9f215fbd9266ca604d487ba331a7febe93a from
https://github.com/Xilinx/linux-xlnx.git

Passing the device tree values to cursor height and width, which was
wrongly set to 0 before.

Signed-off-by: Saurabh Sengar <saurabh.singh@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_mixer.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_mixer.c b/drivers/gpu/drm/xlnx/xlnx_mixer.c
index bcc2bfbceb0b..95a70d54f3f1 100644
--- a/drivers/gpu/drm/xlnx/xlnx_mixer.c
+++ b/drivers/gpu/drm/xlnx/xlnx_mixer.c
@@ -2109,10 +2109,9 @@ static int xlnx_mix_plane_create(struct device *dev, struct xlnx_mix *mixer)
 	mixer->max_width = XVMIX_DISP_MAX_WIDTH;
 	mixer->max_height = XVMIX_DISP_MAX_HEIGHT;
 	if (mixer->hw_logo_layer) {
-		mixer->max_cursor_width =
-			mixer->hw_logo_layer->mixer_layer->layer_regs.width;
-		mixer->max_cursor_height =
-			mixer->hw_logo_layer->mixer_layer->layer_regs.height;
+		layer_data = &mixer_hw->layer_data[XVMIX_LOGO_LAYER_IDX];
+		mixer->max_cursor_width = layer_data->hw_config.max_width;
+		mixer->max_cursor_height = layer_data->hw_config.max_height;
 	}
 	return 0;
 }
-- 
2.31.1

